### PR TITLE
Introduced BaseItemSmallTrashable

### PR DESCRIPTION
**Pull request by Peptide90**
Changed the focus of this PR. Originally it was a baseitem cartridge but then realised it could more generally be a baseitemsmall so did that instead.

Changes:

- Removed cartridge tag as introduced in previous PR in favor of just using the trash tag for things that go in a trash bag.
- In places where I added a tag for trash that didn't have a tag previously, I've parented to the new baseitemsmall so it's inherited rather than duplicated.
- BaseItemSmall has a size of 1 which solves the issue this PR set out to. Ideally cartridges should be smaller than size 1 but this is something to cover another time.
- Moved some other things to baseitemsmall where they benefit from having a size 1 or getting eaten by trash bags.
- Fixed some instances of items that had their own tags not carrying the trash tag.

### CODE REVIEW

File: Resources/Prototypes/Entities/Objects/Weapons/Guns/Ammunition/Cartridges/shotgun.yml
Comment (ShadowCommander) [inline Resources/Prototypes/Entities/Objects/Weapons/Guns/Ammunition/Cartridges/shotgun.yml:10]: Tags don't merge with parent, so this is needed still.

**Review by ShadowCommander — CHANGES_REQUESTED** [review]
Missed the Cartridge tag in `antimaterial.yml`.

File: Resources/Prototypes/Entities/Objects/Weapons/Guns/Ammunition/Cartridges/shotgun.yml
Comment (Peptide90) [reply Resources/Prototypes/Entities/Objects/Weapons/Guns/Ammunition/Cartridges/shotgun.yml:10]: Thought this might be the case, thanks. Luckily I've got a new idea for this PR so happy it didn't get merged.

File: Resources/Prototypes/Entities/Objects/base_item.yml
Comment (ShadowCommander) [inline Resources/Prototypes/Entities/Objects/base_item.yml:42]: Might want to make this BaseItemSmallTrashable, since it might be mistakenly put on other items like pens.

File: Resources/Prototypes/Entities/Objects/Misc/paper.yml
Comment (ShadowCommander) [inline Resources/Prototypes/Entities/Objects/Misc/paper.yml:29]: Not sure if written paper should be trashable.

**Review by ShadowCommander — CHANGES_REQUESTED** [review]

File: Resources/Prototypes/Entities/Objects/base_item.yml
Comment (Peptide90) [reply Resources/Prototypes/Entities/Objects/base_item.yml:42]: Aight, I was actually going to put it on pens but they already had a size component so didn't. Will correct

**Comment by metalgearsloth** [comment]
Next maintainer meeting has prototype composition as a topic so I'd rather wait for that as this seems 1 step closer to SS13's inheritance spaghet.

**Comment by Peptide90** [comment]
> Next maintainer meeting has prototype composition as a topic so I'd rather wait for that as this seems 1 step closer to SS13's inheritance spaghet.

No problemo.

I've got another PR with base item changes in too related to damage able and flammable. It's an odd topic to avoid duplicating code everywhere but then figuring out what to inherit. Hopefully I'm around for the meeting.

File: Resources/Prototypes/Entities/Objects/Tools/flare.yml
Comment (Peptide90) [reply Resources/Prototypes/Entities/Objects/Tools/flare.yml]: Removed for now and added TODO.

File: Resources/Prototypes/Entities/Objects/Misc/paper.yml
Comment (Peptide90) [reply Resources/Prototypes/Entities/Objects/Misc/paper.yml:29]: Put paper back to baseitem since it already had the item size. Needs paper bin in future.

**Comment by Peptide90** [comment]
Closing as moving to composition

**Pull request closed without merging**
